packages/app/src/containers/ComponentsWindow.tsx

```diff
@@ -0,0 +1,91 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import { memo, useCallback } from 'react';
+import ColorSwatch from '../components/swatches/ColorSwatch';
+import ColorSwatchGrid from './ColorSwatchGrid';
+import {
+  useApplicationState,
+} from '../contexts/ApplicationStateContext';
+import {
+  getSharedSwatches
+} from 'noya-state/src/selectors';
+import styled from 'styled-components';
+
+/**
+ * TODO: Multiselect and change color 
+ * TODO: Clean code
+ */
+
+const SwatchContainer = styled.div`
+  height: 175px,
+  &:hover {
+      cursor: pointer;
+  }
+`
+
+
+
+export default memo(function ComponentsWindow() {
+    const [state, dispatch] = useApplicationState();
+
+    let sharedSwatches: Sketch.SwatchContainer | never[] = getSharedSwatches(state);
+
+    let swatches: Array<Sketch.Swatch> = []
+    if (sharedSwatches){
+        sharedSwatches = sharedSwatches as Sketch.SwatchContainer
+        swatches = sharedSwatches.objects;
+    }
+
+    const handleMouseDown = useCallback(
+      (event: React.PointerEvent, swatch?: Sketch.Swatch) => {  
+        switch (state.interactionState.type) {
```

Comment:
We can use `event.stopPropagation()` and `event.preventDefault()` to fix text being selection when shift-clicking.

Here's an example of how we handle the event in list views:

https://github.com/noya-app/noya/blob/main/packages/noya-designsystem/src/components/ListView.tsx#L127-L134

And how to handle shift select in the layer list (although yours will be simpler, since we don't need to handle meta key or handle selecting ranges of items):

https://github.com/noya-app/noya/blob/main/packages/app/src/containers/LayerList.tsx#L99-L126

Reply:
Click events are done in a more Reactive way 🥳